### PR TITLE
Disable caching when node has state.

### PR DESCRIPTION
**Pull request by ayzhuang**

### CODE REVIEW

File: src/ngraph/graph_util.hpp [hunk 1]
```diff
@@ -292,6 +292,9 @@ namespace ngraph
     // the output of this node with in-place kernels
     bool possibly_overwritten(Node* node);
 
+    // Return true if node has state.
```
Comment (diyessi) [inline src/ngraph/graph_util.hpp]: Let's make this a virtual with a default definition in node that returns false.

Comment (ayzhuang) [reply src/ngraph/graph_util.hpp]: If we add m_has_state and has_state() to Node, then we don't need has_state here. cpu_external_function could just do node->has_state().

File: src/ngraph/graph_util.cpp [hunk 2]
```diff
@@ -27,6 +27,7 @@
 #include "ngraph/node.hpp"
 #include "ngraph/op/broadcast.hpp"
 #include "ngraph/op/constant.hpp"
+#include "ngraph/op/experimental/generate_mask.hpp"
```
Comment (aprocter) [inline src/ngraph/graph_util.cpp]: I don't think we need to add this #include.

Comment (ayzhuang) [reply src/ngraph/graph_util.cpp]: will remove. thanks.

**Pull request merged**